### PR TITLE
Properly list all tables in memory connector

### LINKED ISSUE #3195: Presto view in memory connector isn't listed in system.jdbc.tables
Steps to reproduce:
```sql
presto> create view memory.default.view1 as select 1 as c1;
CREATE VIEW

presto> select * from system.jdbc.tables where table_cat = 'memory' and table_schem = 'default';
 table_cat | table_schem | table_name | table_type | remarks | type_cat | type_schem | type_name | self_referencing_col_name | ref_generation
-----------+-------------+------------+------------+---------+----------+------------+-----------+---------------------------+----------------
 memory    | default     | customer   | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | orders     | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | lineitem   | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | part       | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | partsupp   | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | supplier   | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | nation     | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
 memory    | default     | region     | TABLE      | NULL    | NULL     | NULL       | NULL      | NULL                      | NULL
(8 rows)
```

information_schema.views is fine. 
```sql
presto> select * from memory.information_schema.views;
 table_catalog | table_schema | table_name | view_definition
---------------+--------------+------------+-----------------
 memory        | default      | view1      | SELECT 1 c1
               |              |            |
               |              |            |
(1 row)
```

### PR DESCRIPTION
**Pull request by wendigo**
Fixes #3195

### CODE REVIEW

File: presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemoryMetadata.java [hunk 1]
```diff
@@ -198,6 +198,18 @@ public void testCreateViewWithReplace()
                 .hasValue("bbb");
     }
 
+    @Test
+    public void testCreatedViewShouldBeListedAsTable()
+    {
+        SchemaTableName test = new SchemaTableName("test", "test_view");
```
Comment (kokosing) [inline presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemoryMetadata.java]: s/test/viewName

File: presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemoryMetadata.java [hunk 2]
```diff
@@ -198,6 +198,18 @@ public void testCreateViewWithReplace()
...
+
+        metadata.createSchema(SESSION, "test", ImmutableMap.of());
```
Comment (kokosing) [inline presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemoryMetadata.java]: extract "test" as `schemaName`

File: presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java [hunk 3]
```diff
@@ -141,9 +142,12 @@ public synchronized ConnectorTableMetadata getTableMetadata(ConnectorSession ses
     @Override
     public synchronized List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
     {
-        return tables.values().stream()
-                .filter(table -> schemaName.map(table.getSchemaName()::equals).orElse(true))
-                .map(TableInfo::getSchemaTableName)
+        return concat(
```
Comment (kokosing) [inline presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java]: I would use `ImmutableList.builder()` instead of  `concat`

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**